### ISSUE TITLE
md-select value not populated until you click on the select

### ISSUE THREAD

**Issue by nmaves** [angular/material#2588]
We have a select that is populated by an ajax call.  The value in the select is correct on the page.  The label is just not shown until you click on the select.  Once you do you will see that the correct value is selected.  If you click off of the select the correct label that was preselected is shown.

Here is the select code.

```
            <md-select placeholder="Pick" ng-model="browseController.currentKind" ng-change="browseController.reload()">
                <md-option ng-value="kind.id" ng-repeat="kind in browseController.kinds">
                    {{kind.name}}
                </md-option>
            </md-select>
```

And here is the code in the controller to populate the options as well as set the value of the select.

```
        KindsModel.getKinds()
            .then(function (kinds) {
                browseController.kinds = kinds;
                browseController.currentKind = $stateParams.kindId;
            });
```

**Comment by ngraef** [comment]
Duplicate of #2575, though this is a good use case to test.

**Comment by rschmukler** [comment]
Fixed by #2575

**Comment by wembernard** [comment]
Not fixed by #2575

I finally realised if `md-option`(s) are not resolved, placeholder won't display which makes `md-on-open` a broken feature.

Here is a [demo](http://codepen.io/wembernard/pen/garLQK) that show the problem.
